src/main/java/wooteco/subway/service/SectionService.java

```diff
@@ -0,0 +1,112 @@
+package wooteco.subway.service;
+
+import org.springframework.stereotype.Service;
+import wooteco.subway.controller.request.SectionInsertRequest;
+import wooteco.subway.dao.SectionDao;
+import wooteco.subway.domain.Section;
+import wooteco.subway.domain.Sections;
+import wooteco.subway.domain.SimpleSection;
+import wooteco.subway.exception.section.DeleteSectionIsNotPermittedException;
+import wooteco.subway.exception.section.NoneOfSectionIncludedInLine;
+import wooteco.subway.exception.section.SectionDistanceMismatchException;
+import wooteco.subway.exception.section.SectionsAlreadyExistException;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class SectionService {
+
+    private final SectionDao sectionDao;
+
+    public SectionService(SectionDao sectionDao) {
+        this.sectionDao = sectionDao;
+    }
+
+    public void insert(Long lineId, SimpleSection section) {
+        sectionDao.insert(lineId, section);
+    }
+
+    public List<Section> findAllByLineId(Long id) {
+        return sectionDao.findAllByLineId(id);
+    }
+
+    public void validateEndStationsAreIncluded(Long lineId, SectionInsertRequest sectionInsertRequest) {
+        if (sectionDao.isIncludeAllEndStations(lineId, sectionInsertRequest)) {
+            throw new SectionsAlreadyExistException();
+        }
+    }
+
+    public int getSectionCountsByLineId(Long lineId) {
+        return sectionDao.countsByLineId(lineId);
+    }
+
+    public void insertSections(Long lineId, SectionInsertRequest sectionInsertRequest) {
+        final Optional<Section> optionalSectionConversed =
+                sectionDao.findOneIfIncludeConversed(lineId, sectionInsertRequest);
+        if (optionalSectionConversed.isPresent()) {
+            sectionDao.insert(lineId, sectionInsertRequest.toSimpleSectionDto());
+            return;
+        }
+
+        final Optional<Section> optionalSection = sectionDao.findOneIfInclude(lineId, sectionInsertRequest);
+        final Section section = optionalSection.orElseThrow(NoneOfSectionIncludedInLine::new);
+        validateCanBeInserted(section, sectionInsertRequest);
+        final SimpleSection updatedSection = updateOriginalSectionToMakeOneLine(section, sectionInsertRequest);
+        sectionDao.update(lineId, updatedSection); // 기존 섹션을 업데이트함. 삽입된 구간을 포함하여.
+        sectionDao.insert(lineId, sectionInsertRequest.toSimpleSectionDto()); // 추가된 섹션을 삽입함.
+    }
+
+    private SimpleSection updateOriginalSectionToMakeOneLine(Section section, SectionInsertRequest insertedSection) {
+        if (section.getUpStationId().equals(insertedSection.getUpStationId())) {
```

Comment:
오랜만에 보는 객체에게 메세지를 보내라! 
레벨 1 에서 진행했던 부분을 기억해주세요 :) 
```suggestion
        if (section.isEqualsUpStationId(insertedSection.getUpStationId())) {
```